src/circleci/rollcage/ring_middleware.clj

```diff
@@ -0,0 +1,12 @@
+(ns circleci.rollcage.ring-middleware
+  (:require [circleci.rollcage.core :as rollcage]))
+
+(defn wrap-rollbar [handler rollcage-client]
+  (if-not rollcage-client
+    handler
+    (fn [req]
+      (try
+        (handler req)
+        (catch Exception e
+          (rollcage/error rollcage-client e (select-keys req [:uri]))
+          (throw e))))))
```

Comment:
I would have to disagree. I don't think this kind of control flow is uncommon when it comes to ring middleware, or higher order functions in general.

It's hardly a performance improvement, but isn't it still better to have the conditional logic evaluate once when the handler is created, as opposed to each time the returned handler is invoked?